transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -184,6 +187,29 @@ def reset_scan_stats(self):
         self._device_scan_counts = {}
         self._last_reset_time = time.monotonic()
 
+    def _update_conn_map(self, conn_string, device_uuid):
+        """
+        Updates the instance's mapping of device UUIDs to connection strings.
+
+        This uses a simple implementation of a Least Recently Used algorithm.
+        Everytime an entry is updated, it will remove the existing entry
+        and reinstert it at the end. As a result, the least recently used entries
+        will be towards the beginning of the list. That way if our list grows
+        too large and needs to be overwritten, it will do so by removing the
+        beginning elements.
+
+        Args:
+            conn_string (str): The string of the device's MAC address
+            device_uuid (int): The integer representation of the device's UUID
+        """
+        if conn_string in self._conn_map:
+            self._conn_map.move_to_end(conn_string, last=True)
+        else:
+            self._conn_map[conn_string] = device_uuid
+
+        while len(self._conn_map) >= self.ConnMapMaxSize:
```

Comment:
Not super important but the behavior with this order of operations is not super intuitive.  Consider if `ConnMapMaxSize == 1`.  It would result in never filling the conn_map even though the user specified a size of 1.

If you put the `while` at the head of the function you free up space and then add a new entry.  Alternatively the conditional should be `>` not `>=` if you have it in the current location.

Reply:
I'll just change it to `>` at the current location. 
The reason I chose to put that at the end was because I wanted to make sure that we don't pop the oldest value in the scenario that we aren't adding a new entry, but just moving an entry to the end.
The alternative to fix that logic with having it at the head of the function would require us to do a `while len(self._conn_map) > self.ConnMapMaxSize and conn_string not in self._conn_map`.